### PR TITLE
Rework device matching | Store members in customData | Add AC Unit & Extended Fan

### PR DESCRIPTION
**Pull request by michikrug**
⚠️ **Breaking Changes**

- As the group members are now stored in customData, existing (group) devices will only work after a new SYNC
- The `modes` configuration for Thermostats has been renamed to `thermostatModes` - users will have to adjust their configuration
- The `speeds` configuration for Fans has been renamed to `fanSpeeds` - users will have to adjust their configuration
- As item types of group members will now be validated, you might see devices disappearing in Google Home if you used an unsupported item within a group device

⚙️ **Changes**

- Item names for group members are stored in customData to save a query request to openHAB when executing commands  (will supersede #222)
  - For commands that need the current state the query request is still needed (thermostat commands or relative volume)
  - Also some other configuration options are stored in customData
  - Item types of group members will now be validated
  - There is no change on the user's side required but they **will need to do a SYNC**
- Add a lot of new functionality to Fan devices (AirPurifier, Fan, Hood) (will supersede #221)
  - OnOff (separate power switch if used as a group)
  - FanSpeed (separate fan speed if used as a group)
  - Modes (yes, for the first time modes are available)
  - SensorState (for FilterLifeTime and PM2.5 - no clue how to query those)
- Add AC Unit device (will supersede #347 & #358)
  - Basically just a combination of the extended Fan and the Thermostat with all control and config options of both
 
⁉️ **Potential Issues**

As customData is limited to 512 bytes it could be an issue to store very long item names e.g. for devices with a lot of members like thermostat (ref. [developers.google.com/assistant/smarthome/reference/intent/sync](https://developers.google.com/assistant/smarthome/reference/intent/sync))

### CODE REVIEW

**Comment by mbronk** [comment]
Hi @michikrug - I accidentally found this PR, when looking for support for `action.devices.types.AC_UNIT`. 
Do you still plan to pursue these changes?

**Comment by michikrug** [comment]
As I am running those changes for months already in my setup, I feel pretty confident.
I still want to make an official announcement in the openHAB community forum so that people will be aware of those breaking changes.

**Comment by michikrug** [comment]
Sorry for the delay here. Currently being busy with work and private stuff.
Also trying to update my own setup to OH4, which needs a dist upgrade on my PI...

**Comment by michikrug** [comment]
I am super sorry.
I completely lost track of following up the release of this during the last months.

In general it is ready to go for quite some while already.
Due to some breaking changes and time to adjust on the users' side, a slow rollout was planned... but not as slow as it happened now 😁 

I will make another announcement in the community forum in the next days to release this as soon as possible. With hopefully no friction.

Fingers crossed.

**Comment by michikrug** [comment]
Release scheduled for May 31st, 9PM CEST
https://community.openhab.org/t/openhab-google-assistant-v4-release-scheduled/156228
